src/model/storage.py

```diff
@@ -14,57 +14,60 @@
 
 import logging
 
-from cobra.io import read_sbml_model
+import requests
+from cobra.io.dict import model_from_dict
 
 from model.app import app
 
 
 logger = logging.getLogger(__name__)
 
 
-class ModelMeta:
-    """A metabolic model, with metadata and an internal (lazy-loaded in development) cobrapy model instance."""
+class ModelWrapper:
+    """A wrapper for a cobrapy model with some additional metadata."""
 
-    def __init__(self, model_id, species, namespace, biomass_reaction):
-        self.model_id = model_id
-        self.species = species
-        self.namespace = namespace
+    def __init__(self, model, organism_id, biomass_reaction):
+        self.model = model
+        # Use the cplex solver for performance
+        self.model.solver = 'cplex'
+        self.organism_id = organism_id
         self.biomass_reaction = biomass_reaction
 
-        # Preload the model into memory only in the following environments
-        if app.config['ENVIRONMENT'] in ('production', 'staging'):
-            self._load()
 
-    @property
-    def model(self):
-        if not hasattr(self, '_model'):
-            self._load()
-        return self._model
-
-    def _load(self):
-        logger.info(f"Loading model {self.model_id} from SBML file")
-        self._model = read_sbml_model(f"data/models/{self.model_id}.sbml.gz")
-
-        # Use cplex solver
-        self._model.solver = 'cplex'
-
-
-MODELS = [
-    ModelMeta('iJO1366', 'ECOLX', 'bigg', 'BIOMASS_Ec_iJO1366_core_53p95M'),
-    ModelMeta('iMM904', 'YEAST', 'bigg', 'BIOMASS_SC5_notrace'),
-    ModelMeta('iMM1415', 'CRIGR', 'bigg', 'BIOMASS_mm_1_no_glygln'),
-    ModelMeta('iNJ661', 'CORGT', 'bigg', 'BIOMASS_Mtb_9_60atp'),
-    ModelMeta('iJN746', 'PSEPU', 'bigg', 'BIOMASS_KT_TEMP'),
-    ModelMeta('e_coli_core', 'ECOLX', 'bigg', 'BIOMASS_Ecoli_core_w_GAM'),
-    ModelMeta('ecYeast7', 'YEAST', 'yeast7', 'r_2111'),
-    ModelMeta('ecYeast7_proteomics', 'YEAST', 'yeast7', 'r_2111'),
-]
+# Keep all loaded models in memory in this dictionary, keyed by our internal
+# model storage primary key id.
+_MODELS = {}
 
 
 def get(model_id):
-    """Return the meta instance for the given model id"""
-    for model_meta in MODELS:
-        if model_meta.model_id == model_id:
-            return model_meta
-    else:
-        raise KeyError(f"No model with id '{model_id}'")
+    """Return a ModelWrapper instance for the given model id"""
+    if model_id not in _MODELS:
+        _load_model(model_id)
+    return _MODELS[model_id]
+
+
+def _load_model(model_id):
+    logger.debug(f"Requesting model {model_id} from the model warehouse")
```

Comment:
Not a big deal but with the logger the old formatting and arguments should be preferred because in your statement above a new string is constructed every time whereas when you write:

```python
logger.debug("Requesting model %s from the model warehouse", model_id)
```

that will only happen when the logging level is actually debug.